### PR TITLE
feat: implement custom form field validation

### PR DESCRIPTION
**Pull request by agnieszkastanczuk**
## Description 📝

This PR adds custom form field validation.

## Related Issues 🔗

JIRA issue: [FSU-278](https://klau.atlassian.net/browse/FSU-278)

## Resources 📚

React Hook Form (https://react-hook-form.com/)

### CODE REVIEW

File: packages/frontend/app/templates/Contact/components/Form/Form.jsx [hunk 1]
```diff
@@ -1,44 +1,92 @@
...
+      <Styled.Form onSubmit={onSubmit(handleForm)}>
         <Styled.InputContainer>
           <Input
             label="Wpisz Twoje imię"
             placeholder="Twoje imię"
             type="text"
-            name="Imię"
-            required
+            state={determineInputState("firstName")}
+            ref={createRef()}
```
Comment (paulgrym) [inline packages/frontend/app/templates/Contact/components/Form/Form.jsx:56]: this line is redundant

File: packages/frontend/app/templates/Contact/components/Form/Form.jsx [hunk 2]
```diff
@@ -1,44 +1,92 @@
...
+            message={errors.firstName ? errors.firstName.message : ""}
           />
           <Input
             label="Wpisz Twój e-mail"
             placeholder="Twój adres e-mail"
             type="email"
-            name="Mail"
-            required
+            state={determineInputState("email")}
+            ref={createRef()}
```
Comment (paulgrym) [inline packages/frontend/app/templates/Contact/components/Form/Form.jsx:75]: this line is redundant

**Review by paulgrym — COMMENTED** [review]
Good job! Validation in Contact form works perfectly 🤩 We should add validation in Virtual Adoption form as well :)

**Review by paulgrym — APPROVED** [review]

**Review by imklau — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**